### ISSUE TITLE
All app deploys fail

### ISSUE THREAD

**Issue by Kinzi** [dokku/dokku#6774]
### Description of problem

"Suddenly" all my app deploys fail. Not quite sure why.

From what I can see, it seems to be related to an outdated plugin?!? How can I figure out which plugin it is?

I ran dokku-update to try to solve the issue, and it updated alright - but then fails to rebuild the apps. 

### Steps to reproduce

Git deploy or ps:rebuild any app triggers the issue.

### dokku report $APP_NAME

```
dokku report APPNAME
-----> uname: Linux dokku 5.15.0-101-generic #111-Ubuntu SMP Tue Mar 5 20:16:58 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
-----> memory:
                      total        used        free      shared  buff/cache   available
       Mem:            7747        5035         336          23        2375        2378
       Swap:           8191         538        7653
-----> disk utilization:
       Filesystem      Size  Used Avail Use% Mounted on
       /dev/sda1       150G  105G   40G  73% /
       /dev/sda15      253M  142K  252M   1% /boot/efi
-----> disk inode utilization:
       Filesystem     Inodes IUsed IFree IUse% Mounted on
       /dev/sda1        9.3M  1.3M  8.0M   14% /
       /dev/sda15          0     0     0     - /boot/efi
-----> docker version:
       Client: Docker Engine - Community
        Version:           26.0.0
        API version:       1.45
        Go version:        go1.21.8
        Git commit:        2ae903e
        Built:             Wed Mar 20 15:17:48 2024
        OS/Arch:           linux/amd64
        Context:           default

       Server: Docker Engine - Community
        Engine:
         Version:          26.0.0
         API version:      1.45 (minimum version 1.24)
         Go version:       go1.21.8
         Git commit:       8b79278
         Built:            Wed Mar 20 15:17:48 2024
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.6.28
         GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
        runc:
         Version:          1.1.12
         GitCommit:        v1.1.12-0-g51d5e94
        docker-init:
         Version:          0.19.0
         GitCommit:        de40ad0
-----> docker daemon info:
       Client: Docker Engine - Community
        Version:    26.0.0
        Context:    default
        Debug Mode: true
        Plugins:
         buildx: Docker Buildx (Docker Inc.)
           Version:  v0.13.1
           Path:     /usr/libexec/docker/cli-plugins/docker-buildx
         compose: Docker Compose (Docker Inc.)
           Version:  v2.25.0
           Path:     /usr/libexec/docker/cli-plugins/docker-compose

       Server:
        Containers: 42
         Running: 38
         Paused: 0
         Stopped: 4
        Images: 63
        Server Version: 26.0.0
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Using metacopy: false
         Native Overlay Diff: true
         userxattr: false
        Logging Driver: json-file
        Cgroup Driver: systemd
        Cgroup Version: 2
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
        Swarm: inactive
        Runtimes: io.containerd.runc.v2 runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
        runc version: v1.1.12-0-g51d5e94
        init version: de40ad0
        Security Options:
         apparmor
         seccomp
          Profile: builtin
         cgroupns
        Kernel Version: 5.15.0-101-generic
        Operating System: Ubuntu 22.04.4 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 4
        Total Memory: 7.566GiB
        Name: dokku
        ID: 7c37003d-6b8a-4715-a691-f0867282acab
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
         File Descriptors: 331
         Goroutines: 398
         System Time: 2024-03-28T10:04:07.095659456Z
         EventsListeners: 2
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false

-----> herokuish version:
       herokuish: v0.7.5
       buildpacks:
         heroku-buildpack-multi     v1.2.0
         heroku-buildpack-ruby      v267
         heroku-buildpack-nodejs    v240
         heroku-buildpack-clojure   v91
         heroku-buildpack-python    v246
         heroku-buildpack-java      v73
         heroku-buildpack-gradle    v39
         heroku-buildpack-scala     v98
         heroku-buildpack-play      v26
         heroku-buildpack-php       v247
         heroku-buildpack-go        v189
         heroku-buildpack-nginx     v25
         buildpack-null             v3
-----> dokku version: dokku version 0.34.0
-----> dokku-event-listener version: 0.15.0build+5268732
-----> dokku-update version: dokku-update 0.7.2
-----> docker-container-healthchecker version: 0.9.0
-----> docker-image-labeler version: 0.6.1build+c6e15a9
-----> git version: git version 2.34.1
-----> lambda-builder version:        0.6.0
-----> netrc version: 0.8.0build+0751c1b
 !     pack binary is not available
-----> plugn version: plugn: 0.13.0build+fd5297a
-----> sigil version: 0.10.1build+e443be0
-----> sshcommand version: sshcommand 0.17.1
-----> dokku plugins:
         00_dokku-standard    0.34.0 enabled    dokku core standard plugin
         20_events            0.34.0 enabled    dokku core events logging plugin
         app-json             0.34.0 enabled    dokku core app-json plugin
         apps                 0.34.0 enabled    dokku core apps plugin
         apt                  0.13.0 enabled    Inject deb packages into dokku based on files in project
         builder              0.34.0 enabled    dokku core builder plugin
         builder-dockerfile   0.34.0 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.34.0 enabled    dokku core builder-herokuish plugin
         builder-lambda       0.34.0 enabled    dokku core builder-lambda plugin
         builder-nixpacks     0.34.0 enabled    dokku core builder-nixpacks plugin
         builder-null         0.34.0 enabled    dokku core builder-null plugin
         builder-pack         0.34.0 enabled    dokku core builder-pack plugin
         buildpacks           0.34.0 enabled    dokku core buildpacks plugin
         caddy-vhosts         0.34.0 enabled    dokku core caddy-vhosts plugin
         certs                0.34.0 enabled    dokku core certificate management plugin
         checks               0.34.0 enabled    dokku core checks plugin
         common               0.34.0 enabled    dokku core common plugin
         config               0.34.0 enabled    dokku core config plugin
         cron                 0.34.0 enabled    dokku core cron plugin
         docker-options       0.34.0 enabled    dokku core docker-options plugin
         domains              0.34.0 enabled    dokku core domains plugin
         enter                0.34.0 enabled    dokku core enter plugin
         git                  0.34.0 enabled    dokku core git plugin
         haproxy-vhosts       0.34.0 enabled    dokku core haproxy-vhosts plugin
         hostname             0.2.0 enabled    Sets the docker hostname option for dokku
         letsencrypt          0.20.3 enabled    Automated installation of let's encrypt TLS certificates
         logs                 0.34.0 enabled    dokku core logs plugin
         mongo                1.36.8 enabled    dokku mongo service plugin
         mysql                1.38.1 enabled    dokku mysql service plugin
         network              0.34.0 enabled    dokku core network plugin
         nginx-vhosts         0.34.0 enabled    dokku core nginx-vhosts plugin
         openresty-vhosts     0.34.0 enabled    dokku core openresty-vhosts plugin
         plugin               0.34.0 enabled    dokku core plugin plugin
         ports                0.34.0 enabled    dokku core ports plugin
         postgres             1.36.4 enabled    dokku postgres service plugin
         proxy                0.34.0 enabled    dokku core proxy plugin
         ps                   0.34.0 enabled    dokku core ps plugin
         redirect             0.9.1 enabled    Plugin for managing application redirects
         redis                1.37.4 enabled    dokku redis service plugin
         registry             0.34.0 enabled    dokku core registry plugin
         repo                 0.34.0 enabled    dokku core repo plugin
         resource             0.34.0 enabled    dokku core resource plugin
         run                  0.34.0 enabled    dokku core run plugin
         scheduler            0.34.0 enabled    dokku core scheduler plugin
         scheduler-docker-local 0.34.0 enabled    dokku core scheduler-docker-local plugin
         scheduler-k3s        0.34.0 enabled    dokku core scheduler-k3s plugin
         scheduler-null       0.34.0 enabled    dokku core scheduler-null plugin
         shell                0.34.0 enabled    dokku core shell plugin
         ssh-keys             0.34.0 enabled    dokku core ssh-keys plugin
         storage              0.34.0 enabled    dokku core storage plugin
         trace                0.34.0 enabled    dokku core trace plugin
         traefik-vhosts       0.34.0 enabled    dokku core traefik-vhosts plugin
=====> APPNAME app-json information
       App json computed selected:    app.json
       App json global selected:      app.json
       App json selected:
=====> APPNAME app information
       App created at:                1670403283
       App deploy source:             git-push
       App deploy source metadata:    5fe70470b2303f0f70c8eb11e0b68a5fee5d5daa
       App dir:                       /home/dokku/APPNAME
       App locked:                    false
=====> APPNAME builder information
       Builder build dir:
       Builder computed build dir:
       Builder computed selected:
       Builder global build dir:
       Builder global selected:
       Builder selected:
=====> APPNAME builder-dockerfile information
       Builder dockerfile computed dockerfile path: Dockerfile
       Builder dockerfile global dockerfile path: Dockerfile
       Builder dockerfile dockerfile path:
=====> APPNAME builder-herokuish information
       Builder herokuish computed allowed: true
       Builder herokuish global allowed: true
       Builder herokuish allowed:
=====> APPNAME builder-lambda information
       Builder lambda computed lambdayml path: lambda.yml
       Builder lambda global lambdayml path: lambda.yml
       Builder lambda lambdayml path:
=====> APPNAME builder-nixpacks information
       Builder nixpacks computed nixpackstoml path: nixpacks.toml
       Builder nixpacks global nixpackstoml path: nixpacks.toml
       Builder nixpacks nixpackstoml path:
       Builder nixpacks computed no cache: false
       Builder nixpacks global no cache: false
       Builder nixpacks no cache:
=====> APPNAME builder-pack information
       Builder pack computed projecttoml path: project.toml
       Builder pack global projecttoml path: project.toml
       Builder pack projecttoml path:
=====> APPNAME buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest-22
       Buildpacks global stack:
       Buildpacks list:
       Buildpacks stack:
=====> APPNAME caddy information
       Caddy image:                   lucaslorentz/caddy-docker-proxy:2.8
       Caddy letsencrypt email:
       Caddy letsencrypt server:      https://acme-v02.api.letsencrypt.org/directory
       Caddy log level:               ERROR
       Caddy polling interval:        5s
       Caddy tls internal:            false
=====> APPNAME ssl information
       Ssl dir:                       /home/dokku/APPNAME/tls
       Ssl enabled:                   true
       Ssl hostnames:                 cms.contentin.io
       Ssl expires at:                Jun 10 05:24:30 2024 GMT
       Ssl issuer:                    C = US, O = Let's Encrypt, CN = R3
       Ssl starts at:                 Mar 12 05:24:31 2024 GMT
       Ssl subject:                   subject=CN = cms.contentin.io
       Ssl verified:                  verified by a certificate authority
=====> APPNAME checks information
       Checks disabled list:          none
       Checks skipped list:           none
       Checks computed wait to retire: 60
       Checks global wait to retire:  60
       Checks wait to retire:
=====> APPNAME docker options information
       Docker options build:          --link dokku.mysql.contentin_db:dokku-mysql-contentin-db
       Docker options deploy:         --link dokku.mysql.contentin_db:dokku-mysql-contentin-db --restart=on-failure:10 -v /var/lib/dokku/data/storage/APPNAME:/uploads
       Docker options run:            --link dokku.mysql.contentin_db:dokku-mysql-contentin-db -v /var/lib/dokku/data/storage/APPNAME:/uploads
=====> APPNAME domains information
       Domains app enabled:           true
       Domains app vhosts:            cms.contentin.io
       Domains global enabled:        true
       Domains global vhosts:         dokku.webmarken.com
=====> APPNAME git information
       Git deploy branch:             main
       Git global deploy branch:      master
       Git keep git dir:              false
       Git rev env var:               GIT_REV
       Git sha:
       Git source image:
       Git last updated at:           1711566321
=====> APPNAME haproxy information
       Haproxy image:                 byjg/easy-haproxy:4.4.0
       Haproxy letsencrypt email:
       Haproxy letsencrypt server:    https://acme-v02.api.letsencrypt.org/directory
       Haproxy log level:             ERROR
=====> APPNAME letsencrypt information
       Letsencrypt active:            true
       Letsencrypt autorenew:         true
       Letsencrypt computed dns provider:
       Letsencrypt global dns provider:
       Letsencrypt dns provider:
       Letsencrypt computed email:    webmarken@gmail.com
       Letsencrypt global email:      webmarken@gmail.com
       Letsencrypt email:
       Letsencrypt expiration:        1717997070
       Letsencrypt computed graceperiod: 2592000
       Letsencrypt global graceperiod:
       Letsencrypt graceperiod:
       Letsencrypt computed lego docker args:
       Letsencrypt global lego docker args:
       Letsencrypt lego docker args:
       Letsencrypt computed server:   https://acme-v02.api.letsencrypt.org/directory
       Letsencrypt global server:
       Letsencrypt server:
=====> APPNAME logs information
       Logs computed max size:        10m
       Logs global max size:          10m
       Logs global vector sink:       papertrail://?endpoint=logs3.papertrailapp.com:52919&encoding[codec]=json
       Logs max size:
       Logs vector global image:      timberio/vector:0.36.1-debian
       Logs vector sink:
=====> APPNAME network information
       Network attach post create:
       Network attach post deploy:
       Network bind all interfaces:          false
       Network computed attach post create:
       Network computed attach post deploy:
       Network computed bind all interfaces: false
       Network computed initial network:
       Network computed tld:
       Network global attach post create:
       Network global attach post deploy:
       Network global bind all interfaces:   false
       Network global initial network:
       Network global tld:
       Network initial network:
       Network static web listener:
       Network tld:
       Network web listeners:                172.17.0.3:5000
=====> APPNAME nginx information
       Nginx access log format:
       Nginx computed access log format:
       Nginx global access log format:
       Nginx access log path:
       Nginx computed access log path: /var/log/nginx/APPNAME-access.log
       Nginx global access log path:  /var/log/nginx/APPNAME-access.log
       Nginx bind address ipv4:
       Nginx computed bind address ipv4:
       Nginx global bind address ipv4:
       Nginx bind address ipv6:
       Nginx computed bind address ipv6: ::
       Nginx global bind address ipv6: ::
       Nginx client max body size:    150m
       Nginx computed client max body size: 150m
       Nginx global client max body size: 1m
       Nginx disable custom config:
       Nginx computed disable custom config: false
       Nginx global disable custom config: false
       Nginx error log path:
       Nginx computed error log path: /var/log/nginx/APPNAME-error.log
       Nginx global error log path:   /var/log/nginx/APPNAME-error.log
       Nginx hsts include subdomains:
       Nginx computed hsts include subdomains: true
       Nginx global hsts include subdomains: true
       Nginx hsts max age:
       Nginx computed hsts max age:   15724800
       Nginx global hsts max age:     15724800
       Nginx hsts preload:
       Nginx computed hsts preload:   false
       Nginx global hsts preload:     false
       Nginx hsts:
       Nginx computed hsts:           true
       Nginx global hsts:             true
       Nginx last visited at:
       Nginx nginx conf sigil path:
       Nginx computed nginx conf sigil path: nginx.conf.sigil
       Nginx global nginx conf sigil path: nginx.conf.sigil
       Nginx proxy buffer size:
       Nginx computed proxy buffer size: 4k
       Nginx global proxy buffer size: 4k
       Nginx proxy buffering:
       Nginx computed proxy buffering: on
       Nginx global proxy buffering:  on
       Nginx proxy buffers:
       Nginx computed proxy buffers:  8 4k
       Nginx global proxy buffers:    8 4k
       Nginx proxy busy buffers size:
       Nginx computed proxy busy buffers size: 8k
       Nginx global proxy busy buffers size: 8k
       Nginx proxy read timeout:      120s
       Nginx computed proxy read timeout: 120s
       Nginx global proxy read timeout: 60s
       Nginx underscore in headers:
       Nginx computed underscore in headers: off
       Nginx global underscore in headers: off
       Nginx x forwarded for value:
       Nginx computed x forwarded for value: $remote_addr
       Nginx global x forwarded for value: $remote_addr
       Nginx x forwarded port value:
       Nginx computed x forwarded port value: $server_port
       Nginx global x forwarded port value: $server_port
       Nginx x forwarded proto value:
       Nginx computed x forwarded proto value: $scheme
       Nginx global x forwarded proto value: $scheme
       Nginx x forwarded ssl:
       Nginx computed x forwarded ssl:
       Nginx global x forwarded ssl:
/var/lib/dokku/plugins/available/openresty-vhosts/command-functions: line 64: fn-openresty-allowed-letsencrypt-domains-func-base64: command not found
=====> APPNAME openresty information
       Openresty access log format:
       Openresty access log path:     /var/log/nginx/APPNAME-access.log
       Openresty allowed letsencrypt domains func base64:
       Openresty bind address ipv4:
       Openresty bind address ipv6:   ::
       Openresty client max body size:
       Openresty error log path:      /var/log/nginx/APPNAME-error.log
       Openresty global hsts:         true
       Openresty computed hsts:       true
       Openresty hsts:
       Openresty hsts include subdomains: true
       Openresty hsts max age:        15724800
       Openresty hsts preload:        false
       Openresty image:               dokku/openresty-docker-proxy:0.7.0
       Openresty letsencrypt email:
       Openresty letsencrypt server:  https://acme-v02.api.letsencrypt.org/directory
       Openresty proxy buffer size:   4k
       Openresty proxy buffering:     on
       Openresty proxy buffers:       8 4k
       Openresty proxy busy buffers size: 8k
       Openresty proxy read timeout:  60s
       Openresty underscore in headers: off
       Openresty x forwarded for value: $remote_addr
       Openresty x forwarded port value: $server_port
       Openresty x forwarded proto value: $scheme
       Openresty x forwarded ssl:
=====> APPNAME ports information
       Ports map:                     http:80:5000 https:443:5000
       Ports map detected:            http:80:5000 https:443:5000
=====> APPNAME proxy information
       Proxy computed type:           nginx
       Proxy enabled:                 true
       Proxy global type:             nginx
       Proxy type:
=====> APPNAME ps information
       Deployed:                      true
       Processes:                     1
       Ps can scale:                  true
       Ps computed procfile path:     Procfile
       Ps global procfile path:       Procfile
       Ps procfile path:
       Ps restart policy:             on-failure:10
       Restore:                       true
       Running:                       true
       Status web 1:                  running (CID: 2926d56ade7)
=====> APPNAME registry information
       Registry computed image repo:        dokku/APPNAME
       Registry computed push on release:   false
       Registry computed server:
       Registry global image repo template:
       Registry global push on release:
       Registry global server:
       Registry image repo:
       Registry push extra tags:
       Registry push on release:
       Registry server:
       Registry tag version:
=====> APPNAME resource information
=====> APPNAME scheduler information
       Scheduler computed selected:   docker-local
       Scheduler global selected:     docker-local
       Scheduler selected:
=====> APPNAME scheduler-docker-local information
       Scheduler docker local init process: true
       Scheduler docker local parallel schedule count:
=====> APPNAME scheduler-k3s information
       Scheduler k3s computed deploy timeout:       300s
       Scheduler k3s computed image pull secrets:
       Scheduler k3s computed letsencrypt server:   prod
       Scheduler k3s computed namespace:            default
       Scheduler k3s computed rollback on failure:  false
       Scheduler k3s deploy timeout:
       Scheduler k3s global deploy timeout:         300s
       Scheduler k3s global image pull secrets:
       Scheduler k3s global ingress class:          nginx
       Scheduler k3s global kube context:
       Scheduler k3s global kubeconfig path:        /etc/rancher/k3s/k3s.yaml
       Scheduler k3s global letsencrypt email prod:
       Scheduler k3s global letsencrypt email stag:
       Scheduler k3s global letsencrypt server:     prod
       Scheduler k3s global namespace:              default
       Scheduler k3s global network interface:      eth0
       Scheduler k3s global rollback on failure:    false
       Scheduler k3s image pull secrets:
       Scheduler k3s letsencrypt server:
       Scheduler k3s namespace:
       Scheduler k3s rollback on failure:
=====> APPNAME storage information
       Storage build mounts:
       Storage deploy mounts:         -v /var/lib/dokku/data/storage/APPNAME:/uploads
       Storage run mounts:            -v /var/lib/dokku/data/storage/APPNAME:/uploads
=====> APPNAME traefik information
       Traefik api enabled:           false
       Traefik api vhost:             traefik.dokku.me
       Traefik basic auth password:
       Traefik basic auth username:
       Traefik dashboard enabled:     false
       Traefik image:                 traefik:2.11.0
       Traefik letsencrypt email:
       Traefik letsencrypt server:    https://acme-v02.api.letsencrypt.org/directory
       Traefik log level:             ERROR
       ```

### Output of failing commands after running: dokku trace:off

```shell
dokku ps:rebuild APPNAME
-----> Cleaning up...
-----> Building APPNAME from herokuish
 !     Deprecated: please upgrade plugin to use 'pre-build' plugin trigger instead of pre-build-buildpack
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
-----> Adding BUILD_ENV to build environment...
       BUILD_ENV added successfully
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/build": is a directory: unknown: permission denied
Error: failed to start containers: f440ad23f4601566de0ec00bef6b4b0d3db089aa469417d8a5b6754736cffad2
You cannot attach to a stopped container, start it first
 !     Failure during app build
 !     Retagging old image 5533dd8e14c9 as dokku/APPNAME:latest
 !     App build failed
 !      !     Deprecated: please upgrade plugin to use 'pre-build' plugin trigger instead of pre-build-buildpack
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/build": is a directory: unknown: permission denied
Error: failed to start containers: f440ad23f4601566de0ec00bef6b4b0d3db089aa469417d8a5b6754736cffad2
You cannot attach to a stopped container, start it first
 !     Failure during app build
 !     Retagging old image 5533dd8e14c9 as dokku/APPNAME:latest
 !     App build failed
```

**Comment by Kinzi** [comment]
Update: I disabled all non-core plugin but result is the same.

**Comment by josegonzalez** [comment]
Not sure which, but one of your non-core plugins should be updated as it's using the old pre-build-buildpack trigger.

That aside, please update the herokuish package - there was a faulty release that broke herokuish builds. It's been fixed in 0.7.6. Re-running `dokku-update` should put things back in order.

**Comment by Kinzi** [comment]
Not sure why, but running dokku-update  - again - indeed fixed the issue. Not that I didn't try that before reporting ;)

**Comment by josegonzalez** [comment]
Like I mentioned, it was a bad herokuish release (0.7.5) that you happened to install. When you ran the update again, you pulled a fixed version (0.7.6).
